File: go/internal/booking-service/adapters/repositories/guest.repository.go
package repositories

import (
	"book/internal/booking-service/core/domain/entities"

	"gorm.io/gorm"
)

type GuestRepository struct {
	db *gorm.DB
}

func NewGuestRepository(db *gorm.DB) *GuestRepository {
	return &GuestRepository{
		db: db,
	}
}

func (gr *GuestRepository) Create(guest entities.Guest) error {
	if err := gr.db.Create(guest).Error; err != nil {
		return err
	}

	return nil

}

func (gr *GuestRepository) Get(id uint) entities.Guest {
	return entities.Guest{}
}

File: nestjs/src/booking/adapters/repositories/guest.repository.ts
import GuestEntity from "@/booking/core/domain/entities/guest.entity";
import { IGuestRepository } from "@/booking/core/domain/ports/guest.repository";
import { PrismaService } from "../prisma.service";

export class GuestRepository implements IGuestRepository{
    constructor(private prisma: PrismaService){}
    
    Get(id: Number): GuestEntity {
        throw new Error("Method not implemented.");
    }
    
    async Create(guest: GuestEntity): Promise<number> {
        const data = guest.toJSON();

        const newGuest = await this.prisma.guest.create({
            data: {  
                idNumber: data.document.idNumber,
                documentType: data.document.documentType,
                ...data
            }
        });

        return newGuest.id;
    }
}